editor/controllers/mcq-controller.js

```diff
@@ -70,39 +70,34 @@ angular.module('mcqApp', [])
       $scope.generateTelemetry({type: 'SCROLL', id: 'form', target: {id: 'questionunit-mcq-form', ver: '', type: 'form'}})
     });
     $scope.init = function () {
-      $('.menu .item').tab();
-      if (!ecEditor._.isUndefined($scope.questionEditData)) {
-        var data = $scope.questionEditData.data;
-        $scope.mcqFormData.question = data.question;
-        $scope.mcqFormData.options = data.options;
-        $scope.editMedia = $scope.questionEditData.media;
-        //$scope.mcqFormData.media = $scope.questionEditData.media;
-        if (data.length > 2) {
-          for (var j = 2; j < data.length; j++) {
-            $scope.mcqFormData.options.push({
-              'text': '',
-              'image': '',
-              'audio': '',
-              'isCorrect': false
-            });
-            //$scope.mcqConfiguartion.optionsConfig.push({'isText':true,'isImage':true,'isAudio':true,'isHint':true});
-            $scope.$safeApply();
-          }
-        }
-        if ($scope.mcqFormData.options.length < 2) {
-          $scope.mcqFormData.options.splice(2, 1);
+      ecEditor.addEventListener('org.ekstep.questionunit.mcq:validateform',function(event, callback){
+        var validationRes = $scope.formValidation();
+        callback(validationRes.isValid, validationRes.formData);
+      },$scope);
+
+      ecEditor.addEventListener('org.ekstep.questionunit.mcq:editquestion',$scope.editMcqQuestion,$scope);
+      ecEditor.dispatchEvent("org.ekstep.questionunit:compiled");
+    }
+    $scope.editMcqQuestion = function(event,data){
+      var qdata = data.data;
+      $scope.mcqFormData.question = qdata.question;
+      $scope.mcqFormData.options = qdata.options;
+      $scope.editMedia = qdata.media;
+      if (qdata.length > 2) {
+        for (var j = 2; j < qdata.length; j++) {
+          $scope.mcqFormData.options.push({
+            'text': '',
+            'image': '',
+            'audio': '',
+            'isCorrect': false
+          });
+          $scope.$safeApply();
         }
       }
-      $scope.$parent.$on('question:form:val', function (event) { // eslint-disable-line no-unused-vars
-        if ($scope.formValidation()) {
-          /*if dynamic question assign how many questions are create that count to $scope.mcqFormData.questionCount
-          Or else assign 1*/
-          $scope.mcqFormData.questionCount = 1;
-          $scope.$emit('question:form:valid', $scope.mcqFormData);
-        } else {
-          $scope.$emit('question:form:inValid', $scope.mcqFormData);
-        }
-      })
+      if ($scope.mcqFormData.options.length < 2) {
```

Comment:
Why is this needed? We always have minimum of two options right? Check form validation for minimum 2 options.